packages/core/src/Stepper/Step.tsx

```diff
@@ -77,6 +77,7 @@ const StepNumber = styled.span<{
   text-align: center;
   top: 0;
   width: 20px;
+  z-index: 1;
```

Comment:
Imo we should avoid using z-index if the problem can be solved without it. The css for this component is pretty complex, but I think we could get around by refactoring the component to use css grid instead. What do you think?

Reply:
I changed padding-left instead.